### ISSUE TITLE
`rudof` gives lint:double-link warnings

### ISSUE THREAD

**Issue by VladimirAlexiev** [w3c-cg/awesome-semantic-shapes#21]
https://github.com/w3c-cg/awesome-semantic-shapes/actions/runs/12038380640/job/33563929313#step:5:1
Shows a bunch of `lint:double-link` errors because the "rudof" line is duplicated.
I think we should disable this check: @amivanoff is that possible?

**Comment by amivanoff** [comment]
It's doable by "ignore" statement in markdown, like this on duplicate items:

```markdown
<!--lint ignore double-link-->
- [rudof](https://rudof-project.github.io/rudof)
```

But I'm not sure if we should do this. We have faced this issue before with SHACL/ShEx Jena validators https://github.com/w3c-cg/awesome-semantic-shapes/pull/3#issuecomment-2384608754 and have decided (if I've got it correctly there) to use more specific links to the submodules in a source code repository.

This decision is already documented in the [CONTRIBUTING.md](https://github.com/w3c-cg/awesome-semantic-shapes/blob/main/CONTRIBUTING.md).

Example:

This link https://rudof-project.github.io/rudof with label "rudof" is mentioned in two sections "SHACL Validators" and "ShEx Validators". According to the guide, it should become two links:
 - https://github.com/rudof-project/rudof/tree/master/shacl_validation with the label "rudof shacl validation".
 - https://github.com/rudof-project/rudof/tree/master/shex_validation with the label "rudof shex validation".

This link https://rudof-project.github.io/rudof/cli_usage/convert.html from the section "Shape Conversion Tools" and label "rudof" should become the link:
 - https://github.com/rudof-project/rudof/tree/master/rudof_cli with the label "rudof cli".

We could change this approach to any other, but we must update Jena's items in README.md and CONTRIBUTING.md to preserve consistency.

Maybe it is even better to update the CONTRIBUTING.md first.

**Comment by VladimirAlexiev** [comment]
@amivanoff you make very good points, and I edited the title of this issue. 

However, I'd use links to the documentation (and remove or have links to source as secondary):
- https://rudof-project.github.io/rudof/cli_usage/shacl.html
- https://rudof-project.github.io/rudof/cli_usage/shex.html
- https://rudof-project.github.io/rudof/cli_usage/convert.html 

@jeswr You contributed `rudof`, right? Could you fix this?

**Comment by amivanoff** [comment]
So, besides formatting complications from #16 there are some "semantics" questions:
- ontological grounding and uniform grounds for software classification (relation between an abstract "validator" and a specific "software module" from a particular project)
- information consistency in the list (the same approach applied uniformly to the complex software in any language, like Jena, RDF4J, rudof,...)

**Comment by amivanoff** [comment]
Another open question:
* how to assign globally unique dereferenceable identifiers of the software and its parts

**Comment by VladimirAlexiev** [comment]
@amivanoff do you agree with the sentiment in my last comment?

**Comment by amivanoff** [comment]
Ok, do we need to change Jena's and RDF4J's items to make it a consistent approach?

**Comment by VladimirAlexiev** [comment]
@amivanoff Yes! Could you please do it? I think I did it for `rudof` in #31.
And I see in #25 that you add linter switches to turn on/off various validations: is that documented in CONTRIBUTING.md so we don't have to ask you every time?

Huge THANKS for mastering the awesome tools for us!

**Comment by amivanoff** [comment]
I suggested some further improvements on `rudof` list items in https://github.com/w3c-cg/awesome-semantic-shapes/pull/31. I'd like to address them first and then we'll return here.

**Comment by VladimirAlexiev** [comment]
- I went through #13 and #25, accepted many suggestions, and posted one question.
- Due to https://github.com/sindresorhus/awesome-lint/issues/194 we should use your `ignore doule-link` trick to prevent warnings on known good links
- At the same time, we should continue to provide precise links (with fragments) when appropriate.

> Where are the validator's boundaries on this dependency diagram?

I don't know and it's not the job of the awesome list to provide such levels of detail. All we care is to point to "shacl validator" and "shex validator", maybe also to "cli", and to github: that's more than enough info.

So @amivanoff, can we close this or you'd like to continue the discussion?

**Comment by VladimirAlexiev** [comment]
@amivanoff could you please take a look at https://github.com/w3c-cg/awesome-semantic-shapes/actions/runs/12583909394/job/35072436082 ? I think they're all ok but...

**Comment by amivanoff** [comment]
I've added the #43 PR intended to close this issue. It looks, CI job is OK.

**Comment by amivanoff** [comment]
@VladimirAlexiev could you, please, take a look at PR #43 with fixes which closes this issue, and merge it if possible.
